support-models/src/test/scala/com/gu/support/zuora/api/SerialisationSpec.scala

```diff
@@ -36,17 +36,71 @@ class SerialisationSpec extends AsyncFlatSpec with SerialisationTestHelpers with
     testDecoding[ContactDetails](soldToContact, c => c.deliveryInstructions shouldBe Some("Stick it in the shed"))
   }
 
+  "Subscription" should "serialise if it's a non gift" in {
+    val json = dsSubscriptionData.asJson
+    val expected =
+      parse("""
+        |{
+        |  "RatePlanData" : [
+        |    {
+        |      "RatePlan" : { "ProductRatePlanId" : "12345" },
+        |      "RatePlanChargeData" : [],
+        |      "SubscriptionProductFeatureList" : []
+        |    }
+        |  ],
+        |  "Subscription" : {
+        |    "ContractEffectiveDate" : "2020-12-01",
+        |    "ContractAcceptanceDate" : "2020-12-01",
+        |    "TermStartDate" : "2020-12-01",
+        |    "AutoRenew" : true,
+        |    "InitialTermPeriodType" : "Month",
+        |    "InitialTerm" : 12,
+        |    "RenewalTerm" : 12,
+        |    "TermType" : "TERMED",
+        |    "ReaderType__c" : "Direct",
+        |    "CreatedRequestId__c" : "requestId"
+        |  }
+        |}""".stripMargin).toTry.get
+    json.mapObject(_.mapValues(_.dropNullValues)) should be(expected)
+  }
+
+  it should "serialise if it is a gift" in {
+    val json = dsGiftSubscriptionData.asJson
+    val expected =
+      parse("""
+              |{
+              |  "RatePlanData" : [
+              |    {
+              |      "RatePlan" : { "ProductRatePlanId" : "12345" },
+              |      "RatePlanChargeData" : [],
+              |      "SubscriptionProductFeatureList" : []
+              |    }
+              |  ],
+              |  "Subscription" : {
+              |    "ContractEffectiveDate" : "2020-12-01",
+              |    "ContractAcceptanceDate" : "2020-12-01",
+              |    "TermStartDate" : "2020-12-01",
+              |    "AutoRenew" : false,
+              |    "InitialTermPeriodType" : "Month",
+              |    "InitialTerm" : 3,
+              |    "RenewalTerm" : 12,
+              |    "TermType" : "TERMED",
+              |    "ReaderType__c" : "Gift",
+              |    "RedemptionCode__c" : "gd03-asdfghjq",
+              |    "CreatedRequestId__c" : "requestId",
+              |    "GiftNotificationEmailDate__c" : "2020-12-25"
+              |  }
+              |}""".stripMargin).toTry.get
+    json.mapObject(_.mapValues(_.dropNullValues)) should be(expected)
```

Comment:
getting rid of the nulls was tricky - we drop them in the json printer
https://github.com/guardian/support-frontend/blob/547ca4ea72ba19598165c89abb443ec289b5d9de/support-modules/rest/src/main/scala/com/gu/rest/WebServiceHelper.scala#L142
 but if we only go as far as the json, that doesn't apply.